centinel/primitives/dnslib.py

```diff
@@ -15,7 +15,7 @@ def get_ips(host, nameserver=None, record="A"):
     return lookup_domain(host, nameservers=nameservers, rtype=record)
 
 
-def lookup_domain(domain, nameservers=[], rtype="A", timeout=10):
+def lookup_domain(domain, nameservers=[], rtype="A", timeout=2):
```

Comment:
> I think that 2 seconds should be sufficient, but is not guaranteed to be. Some regions (think Africa) typically have RTTs in the range of 400-500ms. If this is a query over TCP (common for EDNS options), then you have two RTTs for the query, which could easily be over 1 second. It would be odd, but I could totally see congestion issues leading to longer timeouts. In our case, the timeout is defined for the socket, but I think it is possible for the timeout to exceed 2 seconds.

I agree that there are some countries that have RTTs that high, but I don't agree that we should make it so much longer for every country everywhere else. We could do country-specific experiments for cases where RTTs happens to be that high, but for others, I think 2 seconds is pretty reasonable.

> Ultimately, the average or typical case really doesn't matter in this case. If we are going to make that claim that a DNS request is being dropped based on this timeout, we cannot settle for an inconclusive measurement-> 2 second is not conclusive enough for me. In these odd cases, we need to wait longer to differentiate packet drops and longer delays. If performance is a concern, then I also built a parallel version of this, so it really doesn't matter.

2 seconds is not inconclusive in **most** cases. Other cases with higher RTTs can be studied separately. As for concurrency, we don't want to have multiple DNS queries issued at the same time because we are recording pcaps for each URL individually. Doing everything concurrently would make pcaps unclean.